crypto/ffc/ffc_params_generate.c

```diff
@@ -53,8 +53,12 @@ static int ffc_validate_LN(size_t L, size_t N, int type)
 #endif
     } else if (type == FFC_PARAM_TYPE_DSA) {
         /* Valid DSA L,N parameters from FIPS 186-4 Section 4.2 */
-        if (L == 1024 && N == 160)
-            return 80;
+#ifdef FIPS_MODULE
+        /* In fips mode 1024/160 can only be used for verification */
+        if (verify)
```

Comment:
I think duplicating code and using "#else", rather than splitting flow of control across and "#ifdef", is more clear.

Reply:
Separated the entire function..